From 243d4c29660cb2bae9890a042dab3b65b94476e1 Mon Sep 17 00:00:00 2001
From: lukefromdc <lukefromdc@hushmail.com>
Date: Sun, 11 Feb 2018 15:43:46 -0500
Subject: [PATCH 1/2] fix backgrounds and emblems dialog content rendering

Fix https://github.com/mate-desktop/caja/issues/506
*note that it still takes a few seconds for new content to show when switching between patterns, colors, and emblems but now they at least appear
---
 eel/eel-wrap-table.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/eel/eel-wrap-table.c b/eel/eel-wrap-table.c
index 793d2290..9d0c05cc 100644
--- a/eel/eel-wrap-table.c
+++ b/eel/eel-wrap-table.c
@@ -194,6 +194,7 @@ eel_wrap_table_size_request (GtkWidget *widget,
     /* The -1 tells Satan to use as much space as is available */
     requisition->width = -1;
     requisition->height = content_dimensions.height + gtk_container_get_border_width (GTK_CONTAINER (widget)) * 2;
+    gtk_widget_queue_allocate (GTK_WIDGET(widget));
 }
 
 static void

From 7ecfee8043b7f808e88bf1ae66b2d233b2a3b9ef Mon Sep 17 00:00:00 2001
From: lukefromdc <lukefromdc@hushmail.com>
Date: Tue, 13 Feb 2018 01:58:53 -0500
Subject: [PATCH 2/2] backgrounds and emblems dialog: ensure content renders
 immediately

Redraw the contents once and exactly once at intial run of eel_wrap_table_draw to get them to show up immediately.
---
 eel/eel-wrap-table.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/eel/eel-wrap-table.c b/eel/eel-wrap-table.c
index 9d0c05cc..be7cf9a7 100644
--- a/eel/eel-wrap-table.c
+++ b/eel/eel-wrap-table.c
@@ -53,6 +53,7 @@ struct EelWrapTableDetails
 
     guint is_scrolled : 1;
     guint cols;
+    gboolean drawn;
 };
 
 /* Private EelWrapTable methods */
@@ -194,7 +195,6 @@ eel_wrap_table_size_request (GtkWidget *widget,
     /* The -1 tells Satan to use as much space as is available */
     requisition->width = -1;
     requisition->height = content_dimensions.height + gtk_container_get_border_width (GTK_CONTAINER (widget)) * 2;
-    gtk_widget_queue_allocate (GTK_WIDGET(widget));
 }
 
 static void
@@ -253,6 +253,11 @@ eel_wrap_table_draw (GtkWidget *widget,
                                       cr);
     }
 
+    /*Redraw the table once and only once to ensure it is displayed */
+    if (wrap_table->details->drawn == FALSE){
+        gtk_widget_queue_allocate (GTK_WIDGET(widget));
+        wrap_table->details->drawn = TRUE;
+        }
     return FALSE;
 }
 
@@ -799,6 +804,8 @@ eel_wrap_table_new (gboolean homogeneous)
 
     eel_wrap_table_set_homogeneous (wrap_table, homogeneous);
 
+    wrap_table->details->drawn = FALSE;
+
     return GTK_WIDGET (wrap_table);
 }
 
